### PR TITLE
Feat:PRO-5823:Added Alert Banner Functionality

### PR DESCRIPTION
**Pull request by austinjb32**
## What does this change do?

- Added Alert Banner in CmAdmin

## Screenshots/Recording (optional)
image
image

image

image

## Anything Else?

It only support  types : **info, success, danger and warning**

### CODE REVIEW

File: app/views/cm_admin/main/_show_content.html.slim [hunk 1]
```diff
@@ -2,6 +2,8 @@
   == render 'cm_admin/main/top_navbar'
   == render 'cm_admin/main/tabs', via_xhr: defined?(via_xhr) ? via_xhr : false
 .show-page__inner
+  -unless @alerts.empty?
```
Comment (mikevic) [inline app/views/cm_admin/main/_show_content.html.slim:5]:
```suggestion
  - unless @alerts.empty?
```

**Review by mikevic — CHANGES_REQUESTED** [review]

File: lib/cm_admin/view_helpers/field_display_helper.rb [hunk 2]
```diff
@@ -26,6 +28,21 @@ def show_rows(ar_object, rows, col_size: 3)
...
+      def create_alerts(alerts, ar_object)
+        alerts.map do |alert|
+          create_alert(alert, ar_object)
+        end.join.html_safe
+      end
+
+      def create_alert(alert, ar_object)
+        return unless alert.display_if.call(ar_object)
+
+        alert_icon_info = { warning: 'triangle-exclamation', danger: 'triangle-exclamation', success: 'circle-check' }
```
Comment (mikevic) [inline lib/cm_admin/view_helpers/field_display_helper.rb:40]:
```suggestion
        alert_icon_info = { warning: 'triangle-exclamation', danger: 'triangle-exclamation', success: 'circle-check' }
```

Is there a reason why this is called `alert_icon_info`? From what I see, it is the defaults based on the type of alert set

**Review by mikevic — CHANGES_REQUESTED** [review]

File: lib/cm_admin/view_helpers/field_display_helper.rb [hunk 3]
```diff
@@ -26,6 +28,21 @@ def show_rows(ar_object, rows, col_size: 3)
...
+        icon_class = alert_icon_info.fetch(alert.type, 'circle-info')
+        alert.body = alert.body.is_a?(Array) ? alert.body : [alert.body]
```
Comment (anbublacky) [inline lib/cm_admin/view_helpers/field_display_helper.rb:42]: We can mandate body to be array right? Even if it is single or multiple

File: lib/cm_admin/view_helpers/field_display_helper.rb (hunk 2 above)
Comment (anbublacky) [reply lib/cm_admin/view_helpers/field_display_helper.rb:40]: @mikevic I believe the keys are based on the alert type and used.

**Review by anbublacky — CHANGES_REQUESTED** [review]

File: lib/cm_admin/view_helpers/field_display_helper.rb (hunk 3 above)
Comment (austinjb32) [reply lib/cm_admin/view_helpers/field_display_helper.rb]: @anbublacky Please check I have added an argument error if not array

**Comment by mikevic** [comment]
@austinjb32  - Bro, can you also add in the docs with this feature? Thanks!

File: app/views/cm_admin/main/_alert_banner.html.slim [hunk 4]
```diff
@@ -0,0 +1,9 @@
+.alert class="alert-#{alert.type}" *alert.html_attrs
+  .d-flex.justify-content-between.align-items-start
+    .d-flex.justify-content-between.align-items-start.py-1
```
Comment (barath46) [inline app/views/cm_admin/main/_alert_banner.html.slim:3]: @austinjb32 Same here

File: app/views/cm_admin/main/_alert_banner.html.slim [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+      i.fa class="fas fa-#{icon_class} fs-5.ms-2"
+    .d-flex.justify-content-between.align-items-center
```
Comment (barath46) [inline app/views/cm_admin/main/_alert_banner.html.slim:5]: @austinjb32 Same here

**Review by barath46 — CHANGES_REQUESTED** [review]
Requested changes

**Review by mikevic — APPROVED** [review]

**Review by mikevic — APPROVED** [review]

**Pull request merged**